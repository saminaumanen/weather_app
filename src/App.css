@import url('https://fonts.googleapis.com/css?family=Sarabun');

body, h1, p {
  font-family: 'Sarabun', sans-serif;
}

#search{
  background-color: #008CBA; 
  border: none;
  border-radius: 4px;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}



#weather_container{
  margin: 20px;
  padding: 20px;
  border-radius: 4px;
  background-color:#F7F7F7;
}


