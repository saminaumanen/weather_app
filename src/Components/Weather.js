import React, { Component } from 'react';


class Weather extends Component {
    
  render(){

    return(

        <div id="city_container">
            {
               this.props.city && <p>Location: 
                    <span className="value">  {this.props.city}</span>                    
                </p> 
            }
            
            {
                this.props.temperature && <p>Temperature: 
                    <span className="value">  {this.props.temperature}</span>
                </p>
            }

            {
                this.props.description && <p>Conditions:  
                    <span className="value">  {this.props.description}</span>
                </p>
            }

            {
                this.props.icon && <p><img src={this.props.icon} alt="Weather Icon" /></p>
            }       
            
            {
                this.props.error && <p className="weather__error">{this.props.error}</p>
            }

        </div>
    )
}
}

export default Weather;