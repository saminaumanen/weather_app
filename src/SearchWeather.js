import React, { Component } from "react";

class SearchWeather extends Component {
  render(){

    return(
            <form onSubmit = {this.props.searchWeather}>
                <input type="text" id="city" placeholder="City..."/><br></br>
                <button id="searchButton">Search</button>
            </form>
    )
}
}

export default SearchWeather;