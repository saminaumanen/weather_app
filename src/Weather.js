import React, { Component } from 'react';

class Weather extends Component {
  render(){

    return(

        <div>
            {
               this.props.city && <p>Location: 
                    <span className="value">  {this.props.city}</span>                    
                </p> 
            }
            
            {
                this.props.temperature && <p>Temperature: 
                    <span className="value">  {this.props.temperature}</span>
                </p>
            }

            {
                this.props.description && <p>Conditions:  
                    <span className="value">  {this.props.description}</span>
                </p>
            }
        </div>
    )
}
}

export default Weather;