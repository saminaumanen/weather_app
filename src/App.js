import React, { Component } from 'react';
import Weather from "./Weather";
import SearchWeather from "./SearchWeather";
import './App.css';


const Api_Key = "2518dd9873caf4ea9437d2619e19c1b5";

class App extends Component {
  state = {
    city: "",
    temperature: "",
    description: "",
    icon:"",
  }


  getWeather = async (e) => {

    const city = e.target.elements.city.value;
    
    e.preventDefault();   
    const api_call = await fetch(`http://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${Api_Key}`);
    const response = await api_call.json();
    console.log(response);
    if(city ){
      this.setState({
        city: response.name,
        temperature: response.main.temp,
        description: response.weather[0].description,
        icon: "http://openweathermap.org/img/w/" + response.weather[0].icon + ".png"
      })
    }
  }

componentWillMount(){
  localStorage.getItem('city') && this.setState({
    city:JSON.parse(localStorage.getItem('city')),
    temperature:JSON.parse(localStorage.getItem('temperature')),
    description:JSON.parse(localStorage.getItem('description')),
    icon:JSON.parse(localStorage.getItem('icon')),
    isLoading: false
  })
}


componentWillUpdate(nextProps, nextState){
  localStorage.setItem('city',JSON.stringify(nextState.city));
  localStorage.setItem('temperature',JSON.stringify(nextState.temperature));
  localStorage.setItem('description',JSON.stringify(nextState.description));
  localStorage.setItem('icon',JSON.stringify(nextState.icon));
}


  render() {

    return (
      <div id="weather_container">
          <h1>Search for a city</h1>
            <p>insert city name to see temperature</p>
          <SearchWeather searchWeather={this.getWeather} />
          <Weather
            city={this.state.city}
            temperature={this.state.temperature}
            description={this.state.description}
            icon={this.state.icon} />
      </div>
    )
  }
}

export default App;
