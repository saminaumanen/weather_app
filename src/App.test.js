import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import SearchWeather from './SearchWeather';
import Weather from './Weather';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow component test', ()=>{
  shallow(<App/>);
})

it('shallow SearchWeather component test', ()=>{
  shallow(<SearchWeather/>);
})

it('shallow Weather component test', ()=>{
  shallow(<Weather/>);
})

it('renders correctly', ()=>{
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
})