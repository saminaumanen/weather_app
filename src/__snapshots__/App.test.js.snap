// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="weather_container"
>
  <h1>
    Search for a city
  </h1>
  <p>
    insert city name to see temperature
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      id="city"
      placeholder="City..."
      type="text"
    />
    <br />
    <button
      id="searchButton"
    >
      Search
    </button>
  </form>
  <div
    id="city_container"
  >
    
    
    
    
  </div>
</div>
`;
